### PR TITLE
W-12658234: User Story G - Retrieve Tracing Overrides from a File

### PR DESCRIPTION
**Pull request by alopezbizon**

### CODE REVIEW

File: modules/tracing/mule-tracer-configuration-impl/src/main/java/org/mule/runtime/tracing/level/impl/config/FileTracingLevelConfiguration.java [hunk 1]
```diff
@@ -53,30 +54,45 @@ public FileTracingLevelConfiguration(MuleContext muleContext) {
...
     ClassLoaderResourceProvider resourceProvider = new ClassLoaderResourceProvider(getExecutionClassLoader(muleContext));
     try {
       InputStream is = resourceProvider
           .getResourceAsStream(CONFIGURATION_PATH + getPropertiesFileName());
       return loadProperties(is);
     } catch (MuleRuntimeException | IOException e) {
-      LOGGER
-          .info("No tracing level config found in the conf directory. The tracing level will be set to the default: "
-              + DEFAULT_LEVEL);
+      LOGGER.info(String
```
Comment (fsgonz) [inline modules/tracing/mule-tracer-configuration-impl/src/main/java/org/mule/runtime/tracing/level/impl/config/FileTracingLevelConfiguration.java:91]: static import.

File: modules/tracing/mule-tracer-configuration-impl/src/main/java/org/mule/runtime/tracing/level/impl/config/FileTracingLevelConfiguration.java [hunk 2]
```diff
@@ -53,30 +54,45 @@ public FileTracingLevelConfiguration(MuleContext muleContext) {
   @Override
   public TracingLevel getTracingLevel() {
     Properties properties = getTracingLevelProperties();
```
Comment (fsgonz) [inline modules/tracing/mule-tracer-configuration-impl/src/main/java/org/mule/runtime/tracing/level/impl/config/FileTracingLevelConfiguration.java:56]: if possible try to read the file only once (not every time you invoke the trace level), by using an initialise or make it lazy, or do not retrieve the file once you have it. This was from the previous implementation.

**Review by fsgonz — CHANGES_REQUESTED** [review]

File: modules/tracing/mule-tracer-configuration-api/src/main/java/org/mule/runtime/tracing/level/api/config/TracingLevelConfiguration.java [hunk 3]
```diff
@@ -15,4 +17,6 @@
 public interface TracingLevelConfiguration {
 
   TracingLevel getTracingLevel();
+
+  HashMap<String, TracingLevel> getTracingLevelOverrides();
```
[the remark this answers is not in the document]
Comment (alopezbizon) [reply modules/tracing/mule-tracer-configuration-api/src/main/java/org/mule/runtime/tracing/level/api/config/TracingLevelConfiguration.java]: Done.

File: modules/tracing/mule-tracer-configuration-impl/src/main/java/org/mule/runtime/tracing/level/impl/config/FileTracingLevelConfiguration.java (hunk 1 above)
Comment (alopezbizon) [reply modules/tracing/mule-tracer-configuration-impl/src/main/java/org/mule/runtime/tracing/level/impl/config/FileTracingLevelConfiguration.java]: Done.

File: modules/tracing/mule-tracer-configuration-impl/src/main/java/org/mule/runtime/tracing/level/impl/config/FileTracingLevelConfiguration.java (hunk 2 above)
Comment (alopezbizon) [reply modules/tracing/mule-tracer-configuration-impl/src/main/java/org/mule/runtime/tracing/level/impl/config/FileTracingLevelConfiguration.java:59]: I made some changes to achieve this. Please review them. Thanks!

**Review by fsgonz — APPROVED** [review]
Approved. This configuration will be read in W-13190334. The configuration format and the internal representation can be seen in W-12658253 that should be the last one for configuration functionality at least in this phase. Apart from performance testing again.

**Pull request merged**